Update V2 with bootstrap 4.5.x, bug corrected on horizontal filters, upgrade django version and more

pulse-mind opened the pull request:
In this pull request I've done a lot of improvements :

upgrade to bootstrap 4.5.x (with badges)
upgrade to django 2.2.x
upgrade node dependencies
upgrade gulp script
close #768
close #769
close #628
close #746
close #704
close #657
close #624
Work for #475

Please accept this pull request

Commit:
close #768 : On my side with these modifications, the demo can run without error.

Commit:
Upgrade node dependencies and gulp script

Commit:
Upgrade bootstrap alpha6
close #769

Commit:
Set 12px for regular font-size

Commit:
Revert "Set 12px for regular font-size"

This reverts commit 1717af9d

Commit:
prepare for migration to django 2 and bootstrap future upgrade

pulse-mind committed:
Bootstrap v4.0.0 (#1)

Migrate to bootstrap v4 
* migrate to bootstrap v4 beta 3 :
Input groups were rewritten with specific .input-group-{prepend|append} classes. The new approach allows us to support validation styles and messages within input groups, while also adding support for custom selects, custom file inputs, and multiple .form-controls.

form-control-static becomes form-control-plaintext

Adding padding px-2 on fields

Commit:
migrate to bootstrap v4.1

Commit:
migrate to bootstrap v4.1

Commit:
Merge remote-tracking branch 'origin/v2' into v2

Commit:
migrate to bootstrap v4.1

Commit:
migrate to bootstrap v4.1.3

Commit:
migrate to bootstrap v4.2.1

Commit:
migrate to bootstrap v4.3.1

Commit:
migrate to bootstrap v4.4.1

Commit:
migrate to bootstrap v4.5.0

pulse-mind committed:
migrate to bootstrap v4.5.0 (#2)

Co-authored-by: Frédéric Ravetier <frederic.ravetier@ametys.org>

Commit:
migrate to bootstrap v4.5.3

Commit:
migrate to django 2.2.x

Commit:
Reduce size 14px to 13px and h2 - h5

pulse-mind committed:
Addlink button (#3)

* #628 : manage addlink button with a lot of horizontal filters
* Update travis and settings for tests

Commit:
Update djangosuit version

pulse-mind committed:
Migrate to Django3.0.x (#4)

* Update to Django 3.0
* Update password form for bootstrap 4
* Added django 3.0 to travis

Commit:
Update django suit version

Commit:
Merge remote-tracking branch 'origin/v2' into v2

pulse-mind committed:
Migration to Django 3.1.x (#5)

* Migrate to Django 3.1.x
adjust css to keep the compatibility
adjust messages

pulse-mind committed:
Django31 (#6)

* Migrate to Django 3.1.x
adjust css to keep the compatibility
adjust messages
Added color on errorlist when a mandatory field is in error

pulse-mind committed:
Migrate to Django3.1 (#7)

Migrate to Django 3.1.x
* adjust css to keep the compatibility
* adjust messages
* Added color on errorlist when a mandatory field is in error

bertini36 commented:
What is the problem with this PR? Is it breaking something? :thinking:

pulse-mind commented:
> What is the problem with this PR? Is it breaking something? 🤔

I understood that darklow (the author, that did a very good job to build this) does not have time to spend on this project so this is it...

bertini36 commented:
Ok, I understand. Thanks for the info @pulse-mind!

gamesbook commented:
> > What is the problem with this PR? Is it breaking something? thinking
> 
> I understood that darklow (the author, that did a very good job to build this) does not have time to spend on this project so this is it...
Unfortunately, in software, its not enough to build a good product - you also need to maintain it, or find someone else to maintain it.

pulse-mind commented:
If you like you can go to my fork and use it. I am not going to improve it, but I am trying to maintain a part of it. I am still not using the v2 on production due to covid, my customer does not have time for now.

pulse-mind committed:
Migrate to Django 3.2 (#8)

* Upgrade to django 3.2

* Revert database

* update travis, exclude pypy

Commit:
Override content_subtitle that was introduce in django 3.2 and not displayed in Django-suit

tjerkw commented:
I think we should all join efforts in a single new django-suits-maintained branch and merge fixes there, what do you think. 

See:
https://github.com/darklow/django-suit/issues/781
https://github.com/darklow/django-suit/issues/475#issuecomment-856635914

tjerkw commented:
https://github.com/darklow/django-suit/issues/781#issuecomment-859611044

pulse-mind commented:
V2 is great and using last bootstrap version which is great. I've done the efforts to have it working. It seems fine for me and better than the V1!

pulse-mind commented:
And I am agree to have only one branch for V2 in order to share our efforts. 
V1 is too old for everything.

Commit:
Update some dependencies

mireq committed:
Element content-wrap is required for delete template (#10)

Hello,

I am sorry for the delay.
Thank you for your commit!

pulse-mind committed:
upgrade to boostrap 4.6.1 (#11)

Commit:
Fixed deprecated JQuery selector in sortables

Commit:
Upgrade node dependencies : including sass
Update gulp for new usages
Update scss for last sass requirements
Bug corrected in font-size (due to previous bootstrap migration)
Bump version

Commit:
Upgrade to bootstrap 5

Commit:
restore the background color of the inputs after bootstrap5 upgrade

Commit:
restore media-breakpoint-down after bootstrap5 upgrade:

Media query mixins parameters have changed for a more logical approach.
media-breakpoint-down() uses the breakpoint itself instead of the next breakpoint (e.g., media-breakpoint-down(lg) instead of media-breakpoint-down(md) targets viewports smaller than lg).

Commit:
restore text-muted after bootstrap5 upgrade for text-help messages

Commit:
Upgrade to bootstrap5 : .custom-select is now .form-select.

Commit:
Upgrade to bootstrap5 : add bootstrap utilities for text-mute and d-none styles (at least). Added in gulp for simple build

Commit:
Upgrade to bootstrap5 : added background-color for results row

Commit:
added a table and a model in custom view

Commit:
Added suit.css for debug and min version
Load min version if debug (DEBUG=True and INTERNAL_IPS set to have it in template)
Update documentation
Bump version

Commit:
Bug corrected when displaying the addlink. It was working for admin but not for a user that has the add permission

Commit:
Added the possibility to add url params in the menu

pulse-mind committed:
Add github workflow instead of travis.ci (#12)

Add github workflow instead of travis.ci

pulse-mind committed:
heroku deploy (#14)

heroku deploy (#14)

pulse-mind committed:
Django 4.0 (#15)

* Upgrade to Django4.0
* Update to prepare deliver on pypi

Commit:
Update readme to add Django 4

Commit:
Added pypi build

Commit:
Added pypi build in readme

pulse-mind committed:
Change default action top behavior (#16)

* Remove by default the action on top + toggle
* Remove pypy from python version github workflow
* Bump version

Commit:
Upgrade package dependencies

Commit:
Improvements on calendarbox and clockbox to display then with a higher zindex
Upgrade dependencies

Commit:
Update django version in github workflow

Commit:
Remove comment in js

Commit:
Bump version

pulse-mind committed:
Django42 (#17)

* update github workflow
* Bump django version to 4.2

pulse-mind committed:
upgrade requirements
